docs/DeveloperGuide.md

```diff
@@ -1,38 +1,253 @@
 # Developer Guide
 
-## Acknowledgements
+## Table of Contents
+
+1. [Acknowledgements](#acknowledgements)
+2. [Design](#design)
+   - [Architecture](#architecture)
+   - [UI component](#ui-component)
+   - [Parser component](#parser-component)
+   - [Storage component](#storage-component)
+   - [Common classes](#common-classes)
+3. [Implementation](#implementation)
+4. [Appendix: Requirements](#appendix--requirements)
+5. [Appendix: Instructions for manual testing](#appendix--instructions-for-manual-testing)
```

Comment:
These 2 links also do not seem to work when I clicked it, perhaps consider changing the '--' to '-' after 'appendix'?